### PR TITLE
➖ remove getSetting

### LINKED ISSUE #5985: Get rid of the `getSettings`
Last piece of removing vuex-options is get rid of the `getSetting` function

Screenshot 2023-05-15 at 13 14 24

### PR DESCRIPTION
**Pull request by roiLeo**
## PR Type

- [x] Refactoring

## Context

- [x] after #5998 
- [x] Closes #5985

### CODE REVIEW

**Review by Jarsen136 — COMMENTED** [review]
There is already a `DEFAULT_PREFIX` for the urlPrefix, so we do need to set `ksm` again

File: utils/mixins/apiUrlMixin.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
+export default class ApiUrlMixin extends mixins(PrefixMixin) {
   get apiUrl() {
-    const endpoint = getChainEndpointByPrefix(
-      this.$store.getters.currentUrlPrefix
-    )
-    return endpoint || this.$store.getters.getSettings['apiUrl']
+    const endpoint: string =
+      getChainEndpointByPrefix(this.urlPrefix) ||
+      getChainEndpointByPrefix('ksm') ||
+      ''
```
Comment (Jarsen136) [inline utils/mixins/apiUrlMixin.ts:16]:
```suggestion
const endpoint: string =
      getChainEndpointByPrefix(this.urlPrefix) 
```

File: utils/WithKeyring.ts [hunk 2]
```diff
@@ -44,12 +43,8 @@ export default class WithKeyring extends ChainMixin {
       ss58Format:
         correctFormat(ss58Forever) >= 0
           ? correctFormat(ss58Forever)
-          : correctFormat(this.prefixByStore),
+          : correctFormat('0'),
```
Comment (vikiival) [inline utils/WithKeyring.ts:46]:
```suggestion
          : correctFormat('2'),
```

File: utils/WithKeyring.ts
Comment (vikiival) [inline utils/WithKeyring.ts:1]: Do we use this component?

File: utils/mixins/apiUrlMixin.ts [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
 @Component
-export default class ApiUrlMixin extends Vue {
+export default class ApiUrlMixin extends mixins(PrefixMixin) {
   get apiUrl() {
-    const endpoint = getChainEndpointByPrefix(
-      this.$store.getters.currentUrlPrefix
-    )
-    return endpoint || this.$store.getters.getSettings['apiUrl']
+    const endpoint: string =
+      getChainEndpointByPrefix(this.urlPrefix) ||
```
Comment (vikiival) [inline utils/mixins/apiUrlMixin.ts:15]: this should be enough, otherwise we it should throw err.
but can be done in different scope

**Review by vikiival — COMMENTED** [review]
Some clarification

File: utils/WithKeyring.ts
Comment (roiLeo) [reply utils/WithKeyring.ts:1]: yes and no, removing it in new commit

File: utils/mixins/apiUrlMixin.ts (hunk 3 above)
Comment (roiLeo) [reply utils/mixins/apiUrlMixin.ts:15]: I don't understand your comment but ok

**Review by vikiival — APPROVED** [review]

File: utils/mixins/apiUrlMixin.ts (hunk 3 above)
Comment (vikiival) [reply utils/mixins/apiUrlMixin.ts]: That it should throw error if we cannot find url with prefix

**Comment by vikiival** [comment]
**[heavy breathing]**

**Pull request merged**